src/Service.Tests/Unittests/SerializationDeserializationTests.cs

```diff
@@ -247,6 +254,63 @@ public void TestColumnDefinitionNegativeCases()
             Assert.IsFalse(_columnDefinition.Equals(col));
         }
 
+        /// <summary>
+        /// Validates serialization and deserilization of Dictionary containing DatabaseTable
+        /// this is how we serialize and deserialize metadataprovider.EntityToDatabaseObject dict
+        /// </summary>
+        [TestMethod]
+        public void TestDictionaryDatabaseObjectSerializationDeserialization()
+        {
+            InitializeObjects();
+
+            Dictionary<string, DatabaseObject> dict = new() { { "person", _databaseTable } };
+
+            string serializedDict = JsonSerializer.Serialize(dict, _options);
+            Dictionary<string, DatabaseObject> deserializedDict = JsonSerializer.Deserialize<Dictionary<string, DatabaseObject>>(serializedDict, _options)!;
+
+            DatabaseTable deserializedDatabaseTable = (DatabaseTable)deserializedDict["person"];
+
+            Assert.AreEqual(deserializedDatabaseTable.SourceType, _databaseTable.SourceType);
+            Assert.AreEqual(deserializedDatabaseTable.FullName, _databaseTable.FullName);
+            deserializedDatabaseTable.Equals(_databaseTable);
```

Comment:
i saw you removed  the overrides for `Equals`, does that apply here/ what determines whether the two databasetables are equal?

Reply:
currently equals in DatabaseObject only checks for Name and SchemaName, hence in line 275 that will be part of equals, for rest I have added different asserts. 
I also have properties count check. I have a seperate test in file handling equality of DatabaseTable:
![image](https://github.com/Azure/data-api-builder/assets/82844317/80e01033-498e-4c9b-9435-be1baca8c8f6)